shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm

```diff
@@ -397,37 +421,15 @@ - (void)applyFlutterLayer:(const FlutterLayer*)layer {
 
   /// Paths in global logical coordinates that need to be clipped to.
   NSMutableArray* paths = RoundedRectClipsFromMutations(masterClip, mutations);
-
-  // Add / remove path clip views depending on the number of paths.
-  while (_pathClipViews.count > paths.count) {
-    NSView* view = _pathClipViews.lastObject;
-    [view removeFromSuperview];
-    [_pathClipViews removeLastObject];
-  }
-
-  NSView* lastView = self;
-
-  for (size_t i = 0; i < paths.count; ++i) {
-    FlutterPathClipView* pathClipView = nil;
-    if (i < _pathClipViews.count) {
-      pathClipView = _pathClipViews[i];
-    } else {
-      pathClipView = [[FlutterPathClipView alloc] initWithFrame:self.bounds];
-      [_pathClipViews addObject:pathClipView];
-      [lastView addSubview:pathClipView];
-    }
-    pathClipView.frame = self.bounds;
-    [pathClipView maskToPath:(__bridge CGPathRef)[paths objectAtIndex:i]
-                  withOrigin:finalBoundingRect.origin];
-    lastView = pathClipView;
-  }
```

Comment:
We've got two arrays here:
1. _pathClipViews (which we want to update)
2. paths (which we're updating from

Lines 401-406 adjust the length of _pathClipViews down to the length of paths if paths is longer.

Lines 410-423 update existing _pathClipViews (lines 412-413 + 419-422) or add additional ones if paths is longer than _pathClipViews (lines 414-422).

The refactor breaks this into three parts:
* Shorten _pathClipViews if needed
* Add additional _pathClipViews if needed
* Iterate over the whole set of _pathClipViews and set the frame + clip paths.